src/cosmos/CosmosEngine.ts

```diff
@@ -290,12 +291,15 @@ export class CosmosEngine extends CurrencyEngine<
       body: { memo }
     } = decodeTxRaw(txRaw)
 
-    let networkFeeCoin = this.networkInfo.defaultTransactionFee
+    let networkFee = this.networkInfo.defaultTransactionFee?.amount ?? '0'
     if (fee != null) {
       const { amount } = fee
-      networkFeeCoin = safeAddCoins([networkFeeCoin, ...amount])
+      const networkFeeCoin = safeAddCoins([
+        coin('0', this.networkInfo.nativeDenom),
```

Comment:
safeAddCoins will add up coins with the same denom and this forces the calculation to use the native denom instead of some other currency. It's the same behavior as filtering the amount array first